python/hail/docs/tutorials/08-joins.ipynb

```diff
@@ -172,11 +185,33 @@
     }
    },
    "source": [
-    "The magic of keys is that they turn tables into maps: `table[expr]` should naturally refer to the row of `table` that has key the value of `expr`.  Note: if the row is not unique, one such row is chosen arbitrarily.\n",
+    "Let's break this syntax down. \n",
+    "\n",
+    "`t2[t1.a]` is an expression referring to the row of table `t2` with value `t1.a`. So this expression will create a map between the keys of `t1` and the rows of `t2`. You can view this mapping directly:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t2[t1.a].show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we only want the field `x` from `t2`, we can select it with `t2[t1.a].x`. Then we add this field to `t1` with the `anntotate_rows()` method. The new joined table `j` has a field `t2_x` that comes from the rows of `t2`. The tables could be joined, because they shared the same number of keys (1) and the same key type (string). The keys do not need to share the same name. Notice that the key values present in `t2` but not in `t1` don't show up in the final result. This join syntax performs a left join. For other types of joins, see the [join()](https://hail.is/docs/devel/hail.Table.html?highlight=join#hail.Table.join) method.\n",
+    "\n",
+    "The magic of keys is that they turn tables into maps: `table[expr]` should naturally refer to the row of `table` that has a key value of `expr`. Note: if the row is not unique, one such row is chosen arbitrarily.\n",
```

Comment:
I think **maps** should be replaced with dictionary. It will be much clearer to a Python user what the analogy is.